arch-design/delfin/RequirementsDesignSpec.md

```diff
@@ -0,0 +1,282 @@
+## Goal
+This document focuses on overall requirement analysis and design of heterogeneous storage monitoring requirements
+
+
+## Non-Goals
+
+* Storage provisioning 
+* Deployment architecture
+
+## Assumptions and Constraints
+1. All requirements, models and interfaces discuss here are heterogeneous, not specific to a vendor model.
+2. Individual modules and use cases are discussed in different documents and the same will be linked here.
+
+## Motivation and background
+Today’s IT infrastructure spans across multi vendor devices and technologies. Administration of heterogeneous platforms is a challenge to storage admins.  Vendor supplied management software is either vendor locked in or not open source.  SODA foundation has vision to standardize and bring heterogeneous device management using a single platform.
+
+## Requirement Analysis
+
+### Input Requirements
+1. Storage infrastructure devices and resources monitoring
+2. Performance metrics monitoring
+3. Alert reporting
+
+### Feature Requirements
+**Storage Device Monitoring** 
+
+- Registering a storage device for monitoring
+
+- Updating a storage device access information
+
+- Removing a storage device from monitoring
+
+- Provide storage devices  attributes
+
+- Synchronize storage devices  information periodically and on user request
+
+
+
+
+ 
+
+**Storage Resources Monitoring**
+
+- Provide storage resources attributes such as pool, LUN, port, etc Details.
+
+- Synchronize storage resources  attributes periodically and on user request
+
+
+
+**Performance Monitoring**
+
+- Register storage for performance monitoring
+- Report key performance indicators periodically 
+
+**Alert Monitoring** 
+
+- Real time reporting of events/alerts.
+
+- Pull alerts from back end on user request.
+
+- Clear Alert on user request.
+
+
+#### Requirement Analysis
+Analyzed various Open Source and Enterprise monitoring solutions.
+
+
+Conclusion:-
+There is no open source framework for addressing all requirements in one platform.
+
+
+
+#### List of Requirements
+##### Functional Requirements
+* Registering a storage device
+* Removing a storage device
+* Get registered storage device list
+* Updating storage device access information
+* Querying storage device details
+* Querying other resource  Information like (LUN,Pool,Disk, etc)
+* Provide performance metrics of different resources
+* Sync all registered  device collection
+* Sync particular  device  collection
+* Register alert source to receive alerts
+* Report alerts
+* Clear alert
+
+
+##### Non Functional Requirements
+**Performance requirements**
+
+NA
+
+**Security requirements**
+1. Support secured access to delfin APIS
+2. Support secured access to back-ends
+
+**Other non functional requirements (scalability  , HA etc…)**
+NA
+
+
+## Architecture Analysis
+### System Architecture
+![Architecture](./resources/Architecture.jpg)
+
+### Module Architecture
+
+NA
+
+### Architecture Tenets
```

Comment:
Please make it bullet points. Just for better appearance

Reply:
Done